indra/sources/tees/tees_api.py

```diff
@@ -104,6 +116,197 @@ def process_text(text, pmid=None, tees_path=None, python2_path=None):
+    # in a compressed xml document
+    sentences_glob = os.path.join(output_dir, '*-sentences.xml.gz')
+    sentences_filename_candidates = glob.glob(sentences_glob)
+
+    # Make sure there is exactly one such file
+    if len(sentences_filename_candidates) != 1:
+        m = 'Looking for exactly one file matching %s but found %d matches'
+        raise Exception(m % (
+            sentences_glob, len(sentences_filename_candidates)))
+        return None, None, None
+
+    # Read in the sentence segmentation XML
+    sentence_segmentation_filename = sentences_filename_candidates[0]
+    with gzip.GzipFile(sentences_filename_candidates[0], 'r') as f:
+        sentence_segmentations = f.read().decode('utf-8')
+
+    # Create a temporary directory to which to extract the a1 and a2 files from
+    # the tarball
+    tmp_dir = tempfile.mkdtemp(suffix='indra_tees_processor')
+
+    try:
+        # Make sure the tarfile with the extracted events is in shared task
+        # format is in the output directory
+        tarfile_glob = os.path.join(output_dir, '*-events.tar.gz')
+        candidate_tarfiles = glob.glob(tarfile_glob)
+        if len(candidate_tarfiles) != 1:
+            raise Exception('Expected exactly one match for glob %s' %
+                            tarfile_glob)
+            return None, None, None
+
+        # Decide what tar files to extract
+        # (We're not blindly extracting all files because of the security
+        # warning in the documentation for TarFile.extractall
+        # In particular, we want to make sure that the filename doesn't
+        # try to specify a relative or absolute path other than the current
+        # directory by making sure the filename starts with an alphanumeric
+        # character.
+        # We're also only interested in files with the .a1 or .a2 extension
+        tar_file = tarfile.open(candidate_tarfiles[0])
+        a1_file = None
+        a2_file = None
+        extract_these = []
+        for m in tar_file.getmembers():
+            if re.match('[a-zA-Z0-9].*.a[12]', m.name):
+                extract_these.append(m)
+
+                if m.name.endswith('.a1'):
+                    a1_file = m.name
+                elif m.name.endswith('.a2'):
+                    a2_file = m.name
+                else:
+                    assert(False)
+
+        # There should be exactly two files that match these criteria
+        if len(extract_these) != 2 or a1_file is None or a2_file is None:
+            raise Exception('We thought there would be one .a1 and one .a2' +
+                            ' file in the tarball, but we got %d files total' %
+                            len(extract_these))
+            return None, None, None
+
+        # Extract the files that we decided to extract
+        tar_file.extractall(path=tmp_dir, members=extract_these)
+
+        # Read the text of the a1 (entities) file
+        with open(os.path.join(tmp_dir, a1_file), 'r') as f:
```

Comment:
Encoding should probably be made explicit here and all other places, i.e. open as 'rb' and decode after reading depending on encoding used by TEES output files.